### ISSUE TITLE
[Datafusion] Error with CAST: Unsupported SQL type Time

### ISSUE THREAD

**Issue by alamb** [apache/arrow-datafusion#193]
*Note*: migrated from original JIRA: https://issues.apache.org/jira/browse/ARROW-12232

> select cast(timestamp as Time) from foo limit 5;
NotImplemented("Unsupported SQL type Time")

**Comment by alamb** [comment]
Comment from Evan Chan(velvia) @ 2021-04-14T20:47:50.722+0000:
<pre>After discussion for the above PR, seems like the above PR does not really solve this issue, and thus will be amended or left.  A related bug/feature will be filed instead.</pre>

**Comment by velvia** [comment]
I'll leave this ticket for tracking implementing CAST() for Time type, and the discussion from the above PR I'll open a separate issue for `to_timestamp_x`

**Comment by Jefffrey** [comment]
Issue seems resolved as of 406c1087bc16f8d2a49e5a9b05d2a0e1b67f7aa5

```bash
jeffrey:~/Code/arrow-datafusion/datafusion-cli$ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.10s
     Running `/media/jeffrey/1tb_860evo_ssd/.cargo_target_cache/debug/datafusion-cli`
DataFusion CLI v14.0.0
❯ select cast(b.timestamp as time) from (select now() as timestamp) b;
+--------------------+
| b.timestamp        |
+--------------------+
| 11:10:31.506762991 |
+--------------------+
1 row in set. Query took 0.004 seconds.
❯ explain select cast(b.timestamp as time) from (select now() as timestamp) b;
+---------------+---------------------------------------------------------------------------------------+
| plan_type     | plan                                                                                  |
+---------------+---------------------------------------------------------------------------------------+
| logical_plan  | Projection: CAST(b.timestamp AS Time64(Nanosecond))                                   |
|               |   Projection: timestamp, alias=b                                                      |
|               |     Projection: TimestampNanosecond(1668597032907296503, Some("+00:00")) AS timestamp |
|               |       EmptyRelation                                                                   |
| physical_plan | ProjectionExec: expr=[CAST(timestamp@0 AS Time64(Nanosecond)) as b.timestamp]         |
|               |   ProjectionExec: expr=[timestamp@0 as timestamp]                                     |
|               |     ProjectionExec: expr=[1668597032907296503 as timestamp]                           |
|               |       EmptyExec: produce_one_row=true                                                 |
|               |                                                                                       |
+---------------+---------------------------------------------------------------------------------------+
2 rows in set. Query took 0.005 seconds.
❯
```

**Comment by alamb** [comment]
🎉  thanks for checking @Jefffrey
